unet/dataset.py

```diff
@@ -43,18 +90,55 @@ def __getitem__(self, start):
             # Reduce our labels to be from 0 to 12
             labels = labels - 1
 
-
             # Now that we have the np array of our label image, we need to resize it down
             # to the same size as our input image - *but* we must be sure to choose an
             # interpolation that uses nearest neighbor and avoids any kind of averaging
             # since that would be meaningless in a labels approach. This is what order=0
             # is doing below.
             # https://scikit-image.org/docs/dev/api/skimage.transform.html#skimage.transform.resize
             labels = resize(labels, self.img_size, order=0)
-            
+
             # labels is (x, y) and we want labels to be (x, y, 1) in shape
             labels = np.expand_dims(labels, 2)
 
+            # Apply data augmentation if requested
+            if self.apply_augmentation:
+                img, labels = self.data_augmentation(img, labels)
+
+            # Assign out labels
+            X[index] = img
             Y[index] = labels
+
+
+
+        # for index, path in enumerate(batch_input_paths):
+        #     img = load_img(path, target_size=self.img_size)
+        #     X[index] = img
+
+        # for index, path in enumerate(batch_target_paths):
+        #     img = load_img(path)
+        #     # Isolate the red channel as that's our labels
+        #     labels, _, _ = img.split()
+        #     # We need this as an numpy array
+        #     labels = np.array(labels, dtype="uint8")
+        #     # Reduce our space forcibly to 1-13. Anything outside our
+        #     # expected labels get set to 3 for OTHER
+        #     labels[labels > 13] = 3
+        #     # Reduce our labels to be from 0 to 12
+        #     labels = labels - 1
+
+
+        #     # Now that we have the np array of our label image, we need to resize it down
+        #     # to the same size as our input image - *but* we must be sure to choose an
+        #     # interpolation that uses nearest neighbor and avoids any kind of averaging
+        #     # since that would be meaningless in a labels approach. This is what order=0
+        #     # is doing below.
+        #     # https://scikit-image.org/docs/dev/api/skimage.transform.html#skimage.transform.resize
+        #     labels = resize(labels, self.img_size, order=0)
+            
+        #     # labels is (x, y) and we want labels to be (x, y, 1) in shape
+        #     labels = np.expand_dims(labels, 2)
+
+        #     Y[index] = labels
```

Comment:
any reason to keep this if it's commented out?

Reply:
Yeah that's a mistake. Cleared it up.